common/src/main/java/tachyon/network/protocol/RPCMessageEncoder.java

```diff
@@ -42,9 +42,9 @@ protected void encode(ChannelHandlerContext ctx, RPCMessage in, List<Object> out
     long bodyBytes = 0;
     DataBuffer payload = null;
 
-    if (in.hasPayload()) {
-      payload = in.getPayloadDataBuffer();
-      bodyBytes = payload.getLength();
+    if (in instanceof PayloadMessage) {
+      payload = ((PayloadMessage)in).getPayloadDataBuffer();
+      bodyBytes = (payload == null) ? 0 : payload.getLength();
```

Comment:
It seems like the previous code looked a bit cleaner? Will `PayloadMessage` have more functionality in the future?